lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
```

Comment:
```javascript
'use strict';

module.exports = promisify;

function promisify(fn, single) {
  Object.defineProperty(inner, 'name', {
    value: `${fn.name}Promisified`
  });
  return inner;

  function inner() {
    return new Promise((resolve, reject) => {
      // this might be slower than array manipulation, but it might not be.
      fn.call(this, ...arguments, single ? resolve : function(err, value) {
        if (err) reject(err);
        resolve(arguments.length <= 2 ? value : Array.prototype.slice.call(arguments, 1));
      });
    });
  }
}
```